src/stream.cpp

```diff
@@ -70,10 +69,10 @@ void zmq::stream_t::xattach_pipe (pipe_t *pipe_, bool subscribe_to_all_)
 
 void zmq::stream_t::xpipe_terminated (pipe_t *pipe_)
 {
-    outpipes_t::iterator it = _outpipes.find (pipe_->get_routing_id ());
-    zmq_assert (it != _outpipes.end ());
-    _outpipes.erase (it);
+    erase_out_pipe (pipe_);
     _fq.pipe_terminated (pipe_);
+    // TODO router_t calls pipe_->rollback() here; should this be done here as
```

Comment:
the rollback was added here: https://github.com/zeromq/libzmq/pull/2577
I don't think there should be cases where the stream socket has unfinished messages left in the pipe, but it should be harmless to call it.
On the other hand I'm not that confident, given the very small amount of tests for the stream socket...

Reply:
Ok, then maybe we leave it as it is, until more stream tests are added.